src/Symfony/Component/Form/Extension/Core/EventListener/ResizeFormListener.php

```diff
@@ -125,7 +128,7 @@ public function preSubmit(FormEvent $event)
                 if (!$form->has($name)) {
                     $form->add($name, $this->type, array_replace(array(
                         'property_path' => '['.$name.']',
-                    ), $this->options));
+                    ), $this->getOptionsForEntry($value)));
```

Comment:
I just realize that there is a big issue with this solution. Above, in `preSetData()`, the method `getOptionsForEntry()` receives the data in the model format. For example, if we have a collection of `Tag` instances, a `Tag` instance would be passed.

Here, however, we pass the submitted data - a string (the tag name) or an array (multiple tag properties) - which are yet to be transformed back into a `Tag` instance. Consequently, the callback needs to be aware of either format and dynamically handle both. That won't be very intuitive nor easy to use in practice.